llvm/test/CodeGen/AMDGPU/srem.ll

```diff
@@ -39,8 +39,8 @@ define amdgpu_kernel void @srem_i16_7(ptr addrspace(1) %out, ptr addrspace(1) %i
 ; TAHITI-NEXT:    s_waitcnt vmcnt(0)
 ; TAHITI-NEXT:    v_readfirstlane_b32 s0, v0
 ; TAHITI-NEXT:    s_mulk_i32 s0, 0x4925
-; TAHITI-NEXT:    s_lshr_b32 s1, s0, 31
 ; TAHITI-NEXT:    s_ashr_i32 s0, s0, 17
+; TAHITI-NEXT:    s_bfe_u32 s1, s0, 0x1000f
```

Comment:
@arsenm @jayfoad Have we missed a fold here? Is BFE better than SRL?

Reply:
This is worse. The BFE has a bigger encoding because it needs to embed the weird immediate

Reply:
OK - I'll see if I can find the missing fold.